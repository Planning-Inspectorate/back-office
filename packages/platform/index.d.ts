import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
