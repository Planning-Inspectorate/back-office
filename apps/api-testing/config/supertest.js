// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
