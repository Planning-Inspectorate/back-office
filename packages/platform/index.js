export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
