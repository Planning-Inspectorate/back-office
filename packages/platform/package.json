{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
