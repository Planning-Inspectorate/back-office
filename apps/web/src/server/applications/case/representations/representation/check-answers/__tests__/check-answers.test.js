import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../../testing/index.js';
import { representationFixture } from '../../../__fixtures__/representations.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };
const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
	nock('http://test/').get('/applications/1/representations/1').reply(200, representationFixture);
};

describe('Representation address details page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl = '/applications-service/case/1/relevant-representations/check-answers?repId=1';

	describe('GET /applications-service/case/1/relevant-representations/check-answers', () => {
		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should have required fields', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Organisation name');
			expect(element.innerHTML).toContain('Full name');
			expect(element.innerHTML).toContain('Job title');
			expect(element.innerHTML).toContain('Email address');
			expect(element.innerHTML).toContain('Address');
			expect(element.innerHTML).toContain('Preferred contact method');
			expect(element.innerHTML).toContain('Type');
			expect(element.innerHTML).toContain('Under 18');
			expect(element.innerHTML).toContain('Representation on behalf of');
			expect(element.innerHTML).toContain('Agent details');
			expect(element.innerHTML).toContain('Representation');
			expect(element.innerHTML).toContain('Date received');
			expect(element.innerHTML).toContain('Original representation');
		});
	});

	describe('POST /applications-service/case/1/relevant-representations/check-answers', () => {
		describe('and there are errors', () => {
			beforeEach(async () => {
				nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
				nock('http://test/')
					.get('/applications/1/representations/1')
					.reply(200, {
						...representationFixture,
						type: ''
					});
			});

			it('should render the page with errors', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
			});
		});

		describe('and there are no errors', () => {
			beforeEach(async () => {
				nocks();
				nock('http://test/')
					.patch(`/applications/1/representations/1`, {
						status: 'AWAITING_REVIEW'
					})
					.reply(200, { message: 'ok' });
			});

			it('should redirect to the next page', async () => {
				const response = await request.post(baseUrl);

				expect(response?.headers?.location).toEqual(
					'/applications-service/case/1/relevant-representations'
				);
			});
		});
	});
});
