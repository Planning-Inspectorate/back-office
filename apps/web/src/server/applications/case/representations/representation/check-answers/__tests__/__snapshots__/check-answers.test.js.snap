// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Representation address details page GET /applications-service/case/1/relevant-representations/check-answers should render the page and have the lookup fields 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <p class=\\"govuk-body\\"><strong>Project name:</strong> mock title</p><span class=\\"govuk-caption-l\\">Add a representation</span>
            <h1             class=\\"govuk-heading-l\\">Check your answers</h1>
                <dl class=\\"govuk-summary-list govuk-!-margin-bottom-9\\">
                    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Organisation name</dt>
                        <dd                         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\"></dd>
                    </div>
                    <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Full name</dt>
                        <dd                         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">Mrs Sue</dd>
                    </div>
                    <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Job title</dt>
                        <dd                         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\"></dd>
                    </div>
                    <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Email address</dt>
                        <dd                         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">test@example.com</dd>
                    </div>
                    <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Telephone number</dt>
                        <dd                         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">01234 567890</dd>
                    </div>
                </dl>
                </dt>
                <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
                <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/contact-details?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change contact details</span></a>
                </dd>
        </div>
        <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Address</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">96 The Avenue
                <br>Maidstone
                <br>MD21 5XY</dd>
        </div>
        </dl>
        </dt>
        <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/address-details?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change address details</span></a>
        </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Preferred contact method</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">Email</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/contact-method?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change preferred contact method</span></a>
    </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Type</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">Local authorities</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/representation-type?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change type</span></a>
    </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Under 18</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">No</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/under-18?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change under 18</span></a>
    </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Representation on behalf of</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">An individual</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/representation-entity?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change representation on behalf of</span></a>
    </dd>
    </div>
    </dl>
    <h3 class=\\"govuk-heading-m\\">Agent details</h3>
    <dl class=\\"govuk-summary-list govuk-!-margin-bottom-9\\">
        <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Organisation name</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\"></dd>
        </div>
        <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Full name</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">James Bond</dd>
        </div>
        <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Job title</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\"></dd>
        </div>
        <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Email address</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">test-agent@example.com</dd>
        </div>
        <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Telephone number</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">01234 567890</dd>
        </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/contact-details?repId=1&amp;repType=representative&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change contact details</span></a>
    </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Address</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">92 Huntsmoor Road
            <br>RG26 4BX</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/address-details?repId=1&amp;repType=representative&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change address details</span></a>
    </dd>
    </div>
    <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Preferred contact method</dt>
        <dd         class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">Email</dd>
    </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/contact-method?repId=1&amp;repType=representative&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change preferred contact method</span></a>
    </dd>
    </div>
    </dl>
    <h3 class=\\"govuk-heading-m\\">Representation</h3>
    <dl class=\\"govuk-summary-list govuk-!-margin-bottom-9\\">
        <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key govuk-!-padding-0 width--auto\\"><dl class=\\"govuk-summary-list display--block govuk-!-margin-0\\" id=\\"\\"><div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Date received</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">31 May 2023</dd>
        </div>
        <div class=\\"govuk-summary-list__row display--flex flex-wrap--wrap govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key govuk-!-width-one-third\\"> Original representation</dt>
            <dd             class=\\"govuk-summary-list__value govuk-!-width-two-thirds govuk-!-margin-0\\">The proposals size, loss of food producing land, battery storage size
                and technology, joined up thinking between DCO applications</dd>
        </div>
    </dl>
    </dt>
    <dd class=\\"govuk-summary-list__value width--0 govuk-!-padding-0\\"></dd>
    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/1/relevant-representations/add-representation?repId=1&amp;repType=represented&amp;repMode=check\\"> Change<span class=\\"govuk-visually-hidden\\"> Change representation</span></a>
    </dd>
    </div>
    </dl>
    <form method=\\"post\\">
        <button value=\\"1\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\">Submit for review</button>
    </form>
    </div>
    </div>
</main>"
`;
